backtrace-library/src/main/java/backtraceio/library/services/BacktraceDatabaseContext.java

```diff
@@ -175,13 +175,20 @@ public boolean delete(BacktraceDatabaseRecord record) {
         }
 
         for (int key : batchRetry.keySet()) {
-            for (BacktraceDatabaseRecord databaseRecord : batchRetry.get(key)) {
+            List<BacktraceDatabaseRecord> records = batchRetry.get(key);
+
+            for (BacktraceDatabaseRecord databaseRecord : records) {
                 if (databaseRecord == null || record.id != databaseRecord.id) {
                     continue;
                 }
 
                 databaseRecord.delete();
-                batchRetry.get(key).remove(databaseRecord);
+                try {
+                    records.remove(databaseRecord);
+                }
+                catch (Exception e) {
+                    BacktraceLogger.d(LOG_TAG, "Exception on removing record from db context: " + e.getMessage());
+                }
                 this.totalRecords--;
```

Comment:
Are you should we should decrrase number of reports and database size? It looks like it is already handled by someone else

Reply:
I was thinking about it, ok let's do this.